File: rcore/src/main/java/com/singlethread/rcore/utils/SP.java
package com.singlethread.rcore.utils;

import android.content.Context;

import com.singlethread.rcore.SharedPreferenceManager;

import java.util.Map;

/**
 * Created by liurenhui on 15/3/27. SharedPreference 封装
 */
public class SP {

    public String spName;
    private SharedPreferenceManager spManager;

    /**
     * @param name 作为key的前缀
     */
    public SP(Context context, String name) {
        this.spName = name;
        spManager = SharedPreferenceManager.get(context.getApplicationContext());
    }

    public static SP getDefaultSP(Context context) {
        return new SP(context,"Default_SP");
    }

    public static SP getSP(Context context,String spName) {
        return new SP(context,spName);
    }

    private String mappingKey(String key) {
        return spName + "_" + key;
    }

    public String getString(String key, String defaultValue) {
        return spManager.get(mappingKey(key), defaultValue);
    }

    public void putString(String key, String value) {
        spManager.put(mappingKey(key), value);
    }

    public void putInt(String key, int value) {
        spManager.put(mappingKey(key), value);
    }

    public int getInt(String key, int defaultValue) {
        return spManager.get(mappingKey(key), defaultValue);
    }

    public void putLong(String key, long value) {
        spManager.put(mappingKey(key), value);
    }

    public long getLong(String key, long defaultValue) {
        return spManager.get(mappingKey(key), defaultValue);
    }

    public void putFloat(String key, float value) {
        spManager.put(mappingKey(key), value);
    }

    public float getFloat(String key, float defaultValue) {
        return spManager.get(mappingKey(key), defaultValue);
    }

    public void putBoolean(String key, boolean value) {
        spManager.put(mappingKey(key), value);
    }

    public boolean getBoolean(String key, boolean defaultValue) {
        return spManager.get(mappingKey(key), defaultValue);
    }

    public void remove(String key) {
        spManager.remove(mappingKey(key));
    }

    public void clear() {
        spManager.clear();
    }

    public Map<String, ?> getAll() {
        return spManager.getAll();
    }
}

File: lib_base/src/main/java/com/cuke/base/utils/SP.kt
package com.cuke.base.utils

import android.content.Context

class SP(context: Context, var spName: String) {
    private var spManager: SharedPreferenceManager = SharedPreferenceManager.get(context.applicationContext)!!
    private fun mappingKey(key: String): String {
        return spName + "_" + key
    }

    fun getString(key: String, defaultValue: String): String? {
        return spManager.get(mappingKey(key), defaultValue)
    }

    fun putString(key: String, value: String) {
        spManager.put(mappingKey(key), value)
    }

    fun putInt(key: String, value: Int) {
        spManager.put(mappingKey(key), value)
    }

    fun getInt(key: String, defaultValue: Int): Int {
        return spManager.get(mappingKey(key), defaultValue)
    }

    fun putLong(key: String, value: Long) {
        spManager.put(mappingKey(key), value)
    }

    fun getLong(key: String, defaultValue: Long): Long {
        return spManager.get(mappingKey(key), defaultValue)
    }

    fun putFloat(key: String, value: Float) {
        spManager.put(mappingKey(key), value)
    }

    fun getFloat(key: String, defaultValue: Float): Float {
        return spManager.get(mappingKey(key), defaultValue)
    }

    fun putBoolean(key: String, value: Boolean) {
        spManager.put(mappingKey(key), value)
    }

    fun getBoolean(key: String, defaultValue: Boolean): Boolean {
        return spManager.get(mappingKey(key), defaultValue)
    }

    fun remove(key: String) {
        spManager.remove(mappingKey(key))
    }

    fun clear() {
        spManager.clear()
    }

    val all: Map<String, *>
        get() = spManager.all

    companion object {
        fun getDefaultSP(context: Context): SP {
            return SP(context, "Default_SP")
        }

        fun getSP(context: Context, spName: String): SP {
            return SP(context, spName)
        }
    }

}